systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java

```diff
@@ -0,0 +1,591 @@
+        clientsCa = new SystemTestCertHolder(extensionContext,
+            "CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClientsCA",
+            KafkaResources.clientsCaCertificateSecretName(ts.getClusterName()),
+            KafkaResources.clientsCaKeySecretName(ts.getClusterName()));
+
+        //  d) Update the CA certificate.
+        clientsCaCertificateSecret.getData().put("ca.crt", Base64.getEncoder().encodeToString(Files.readAllBytes(Paths.get(clientsCa.getBundle().getCertPath()))));
+
+        //  e) Increase the value of the CA certificate generation annotation.
+        //  f) Save the secret with the new CA certificate and certificate generation annotation value.
+        SystemTestCertHolder.increaseCertGenerationCounterInSecret(clientsCaCertificateSecret, ts, Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION);
+
+        // ------- private key part
+
+        // 5. Update the Secret for the CA key used to sign your new CA certificate.
+        //  a) Edit the existing secret to add the new CA key and update the key generation annotation value.
+        final Secret clientsCaKeySecret = kubeClient(ts.getNamespaceName()).getSecret(ts.getNamespaceName(), KafkaResources.clientsCaKeySecretName(ts.getClusterName()));
+
+        //  b) Encode the CA key into base64.
+        //  c) Update the CA key.
+        final File strimziKeyPKCS8 = SystemTestCertManager.convertPrivateKeyToPKCS8File(clientsCa.getSystemTestCa().getPrivateKey());
+        clientsCaKeySecret.getData().put("ca.key", Base64.getEncoder().encodeToString(Files.readAllBytes(Paths.get(strimziKeyPKCS8.getAbsolutePath()))));
+        // d) Increase the value of the CA key generation annotation.
+        // 6. Save the secret with the new CA key and key generation annotation value.
+        SystemTestCertHolder.increaseCertGenerationCounterInSecret(clientsCaKeySecret, ts, Ca.ANNO_STRIMZI_IO_CA_KEY_GENERATION);
+
+        // --- verification phase (Rolling Update of components)
+
+        // 7. save the current state of the Kafka
+        final Map<String, String> kafkaPods = PodUtils.podSnapshot(ts.getNamespaceName(), ts.getKafkaSelector());
+
+        // 8. Resume reconciliation from the pause.
+        LOGGER.info("Resume the reconciliation of the Kafka custom resource ({}).", KafkaResources.kafkaStatefulSetName(ts.getClusterName()));
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(ts.getClusterName(), kafka -> {
+            kafka.getMetadata().getAnnotations().remove(Annotations.ANNO_STRIMZI_IO_PAUSE_RECONCILIATION);
+        }, ts.getNamespaceName());
+
+        // 9. On the next reconciliation, the Cluster Operator performs a `rolling update` only for the
+        // Kafka cluster. When the rolling update is complete, the Cluster Operator will start a new one to
+        // generate new server certificates signed by the new CA key.
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(ts.getKafkaSelector(), 1, kafkaPods);
```

Comment:
there are no server certificates to generate when you are renewing the clients CA cert and key. Only just one Kafka brokers rolling should happen to trust the new clients CA cert but not rolling twice. The clients CA key is used only by UO to generate the clients certs that will be trusted by brokers via the clients CA cert. Or you see something different?

Reply:
Yes, correct...I think it was copy-pasted error... )